### PR TITLE
Fixing some old issues

### PR DESCRIPTION
**Pull request by IgorEisberg**

### CODE REVIEW

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java [hunk 1]
```diff
@@ -172,15 +172,10 @@ public void recordUncompressedFiles(ExtFile apkFile, Collection<String> uncompre
                     unk.getCompressionLevel(file) == 0 &&
                     unk.getSize(file) != 0) {
 
-                    if (StringUtils.countMatches(file, ".") > 1) {
+                    ext = FilenameUtils.getExtension(file);
+                    if (ext.isEmpty() || (!ext.equals("png") && StringUtils.countMatches(file, ".") > 1)) {
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: Don't know if I feel comfortable with this patch. This happens with plenty of other attributes outside of PNG. I need to research a proper fix which might be detecting two extensions and saving as such.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java [hunk 2]
```diff
@@ -1012,9 +1012,8 @@ public void close() throws IOException {
...
 
     private final static String[] IGNORED_PACKAGES = new String[] {
-            "android", "com.htc", "miui", "com.lge", "com.lge.internal", "yi", "com.miui.core", "flyme",
-            "air.com.adobe.appentry", "FFFFFFFFFFFFFFFFFFFFFF" };
+            "android", "com.htc", "com.lge", "com.lge.internal", "yi", "flyme", "air.com.adobe.appentry",
+            "FFFFFFFFFFFFFFFFFFFFFF" };
 
-    private final static String[] ALLOWED_PACKAGES = new String[] {
-            "com.miui" };
+    private final static String[] ALLOWED_PACKAGES = new String[] { };
```
Comment (iBotPeaches) [inline brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java]: If the reason I added `ALLOWED_PACKAGES` is no longer needed (ie miui). We can cleanup a lot of the codebase as this was a hacky fix I had to do for MIUI back when we actually supported it.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java (hunk 1 above)
Comment (IgorEisberg) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: Sure there might be more extensions to consider, but currently the .9.png extension is the most prominent and is the only extension that spams the apktool.yml file and prevents Windows recompile. I think it's a solid workaround until/if we have a more universal one.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java (hunk 2 above)
Comment (IgorEisberg) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java]: Probably. What I do know is that com.miui package doesn't even exist.
Nowadays the packages are:
miui/com.miui.core
android.miui/com.miui.rom
miui.system/com.miui.system

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java (hunk 1 above)
Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: Probably right there, I'll do some double checks and merge this

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java (hunk 2 above)
Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java]: Thanks, I'll clean it up and remove the workaround. Would help to cut down the codebase.

File: brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java (hunk 1 above)
Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: Sorry for the delay, getting back into dev. It seems `.9.png` images are not compressed. So we should just add support to check for more than one extension and thus store ".9.png" into the list.

Comment (IgorEisberg) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: The problem is that there is virtually no way to distinguish an extra extension from just a part of the file name. A dot is a valid character in a resource name. It's just that <code>.9.png</code> should get the same treatment as <code>.png</code>.

Comment (iBotPeaches) [reply brut.apktool/apktool-lib/src/main/java/brut/androlib/Androlib.java]: I see, maybe we will start with a predefined list to get around the valid character. So we just look from the end of the string for `9.png` and work from there.

**Comment by iBotPeaches** [comment]
Sorry for the delay. These have been cherry-picked, cleaned up and merged. Thanks again.

https://github.com/iBotPeaches/Apktool/commit/8084762274c66e1f91f9625aeb0b5eeccaa09463
https://github.com/iBotPeaches/Apktool/commit/bc9f465f3ec2cf02466dcd2004cbd436439ceed6

**Pull request closed without merging**